src/core/src/bootstrap/EnvLayer.py

```diff
@@ -79,6 +83,79 @@ def get_package_manager(self):
 
         return ret
 
+    def set_env_var(self, var_name, var_value=None, raise_if_not_success=False):
+        """ Sets an environment variable with var_name and var_value in /etc/environment. If it already exists, it is overwriten. """
+        try:
+            environment_vars = self.file_system.read_with_retry(self.etc_environment_file_path)
+            if environment_vars is None:
+                print("Error occurred while setting environment variable: File not found. [Variable={0}] [Value={1}] [Path={2}]".format(str(var_name), str(var_value), self.etc_environment_file_path))
+                return
+
+            environment_vars_lines = environment_vars.strip().split("\n")
+
+            if var_value is None:
+                # remove environment variable
+                if var_name not in environment_vars:
+                    return
+
+                regex = re.compile('{0}=.+'.format(var_name))
+                search = regex.search(environment_vars)
+                if search is None:
+                    return
+
+                group = search.group()
+                environment_vars = environment_vars.replace(group, '').replace("\n\n", "\n")
+                self.file_system.write_with_retry(self.etc_environment_file_path, environment_vars, "w")
+                return
+
+            formatted_env_var = "{0}={1}".format(var_name, str(var_value))
+            if var_name not in environment_vars:
+                self.file_system.write_with_retry(self.etc_environment_file_path, "\n" + formatted_env_var)
+            else:
+                # Update the value of the existing setting
+                for env_var in environment_vars_lines:
+                    if var_name not in str(env_var):
+                        continue
+
+                    environment_vars = environment_vars.replace(str(env_var), formatted_env_var)
+                    break
+
+                self.file_system.write_with_retry(self.etc_environment_file_path, environment_vars, 'w')
+
```

Comment:
Maybe in the next PR, if not enough time here, but the parts for writing when the env_var pre-exists in the the file and when it doesn't could be combined into one.

Refer https://github.com/Azure/LinuxPatchExtension/blob/aaa60bcb59f7a282ac8337a1cf31e9e120bb1e08/src/core/src/package_managers/AptitudePackageManager.py#L483 for a similar example where we write to a config file

Reply:
Got it, thanks.

Reply:
God improvement to the logic but consider having a backlog item to further improve this, for instance we could have a single write_with_retry for 

 if var_name not in environment_vars:
                self.file_system.write_with_retry(self.etc_environment_file_path, "\n" + formatted_env_var)
            else:
                # Update the value of the existing setting